src/app/js/lib/components/SelectField.js

```diff
@@ -0,0 +1,102 @@
+import {
+    FormControl,
+    FormHelperText,
+    InputLabel,
+    MenuItem,
+    Select,
+} from '@mui/material';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import { useTranslate } from '../../i18n/I18NContext';
+
+// TextField component to use tanstack react form with material ui text field
+export function SelectField({
+    form,
+    name,
+    label,
+    helperText,
+    required,
+    options,
+}) {
+    const { translate } = useTranslate();
+    return (
+        <form.Field name={name}>
+            {(field) => {
+                const error = (() => {
+                    // required is used for optionally required field based on a condition
+                    // since tanstack form does not support multi field validation on the field side
+                    if (required) {
+                        return field.state.meta.isTouched && !field.state.value
+                            ? 'error_field_required'
+                            : null;
+                    }
+                    return field.state.meta.isTouched &&
+                        field.state.meta.errors?.length
+                        ? field.state.meta.errors[0]
+                        : null;
+                })();
```

Comment:
How about using useField instead of form.Field, this way you can use a useMemo for the error computation